process/transactionLog/process.go

```diff
@@ -161,25 +162,65 @@ func (tlp *txLogProcessor) SaveLog(txHash []byte, tx data.TransactionHandler, lo
 		})
 	}
 
+	tlp.mut.Lock()
+	defer tlp.mut.Unlock()
+
 	tlp.saveLogToCache(txHash, txLog)
 
-	buff, err := tlp.marshalizer.Marshal(txLog)
+	return tlp.appendLogToStorer(txHash, txLog)
+}
+
+func (tlp *txLogProcessor) appendLogToStorer(txHash []byte, newLog *transaction.Log) error {
+	oldLogsBuff, errGet := tlp.storer.Get(txHash)
+	if isFirstEntryForHash(oldLogsBuff, errGet) {
+		allLogsBuff, err := tlp.marshalizer.Marshal(newLog)
+		if err != nil {
+			return err
+		}
+
+		return tlp.storer.Put(txHash, allLogsBuff)
+	}
+	if errGet != nil {
+		return errGet
+	}
+
+	oldLogs := &transaction.Log{}
+	err := tlp.marshalizer.Unmarshal(oldLogs, oldLogsBuff)
 	if err != nil {
 		return err
 	}
 
-	return tlp.storer.Put(txHash, buff)
+	if oldLogs.Address == nil {
+		oldLogs.Address = newLog.Address
+	}
+	oldLogs.Events = append(oldLogs.Events, newLog.Events...)
```

Comment:
this is super inefficient. until now for a hash we added the logEvent only once when all the execution was completed. Please do that here as well. 

In the current format this PR is not ok.

Reply:
pushed

updated the flow a bit so this only happens for failed inner intra shard transactions. The append functionality is still needed, as for failed intra shard  inner transactions we don't have scrs, but only events on the big relayed tx(key = relayed tx hash).

Reply:
but you could accumulate the logs at the transaction processor level and push here once.

Reply:
pushed.. added a new component able to accumulate all logs during execution, then at the end only one SaveLog is done

also removed AppendLogs functionality